File: internal/extractor/scip/testdata/fixtures/java-spring/src/main/java/example/CampaignController.java
package example;

/**
 * CampaignController is the HTTP entry point. The deleteCampaign
 * method on line 20 is the symbol the integration test resolves.
 * The walker must trace deleteCampaign → CampaignService.softDelete →
 * CampaignRepository.deleteById.
 */
public class CampaignController {

    private final CampaignService service;

    public CampaignController(CampaignService service) {
        this.service = service;
    }

    // Anchor for the integration test: deleteCampaign starts on line 20.
    public void deleteCampaign(String id) {
        if (id == null) {
            throw new IllegalArgumentException("id required");
        }
        service.softDelete(id);
    }
}

File: internal/extractor/scip/testdata/fixtures/kotlin-spring/src/main/kotlin/example/CampaignController.kt
package example

/**
 * CampaignController is the HTTP entry point. The deleteCampaign
 * method (declared around line 13 of this file) is the symbol the
 * integration test resolves. The walker must trace
 * deleteCampaign → CampaignService.softDelete →
 * CampaignRepository.deleteById.
 */
class CampaignController(private val service: CampaignService) {

    // Anchor for the integration test: deleteCampaign is the entry symbol.
    fun deleteCampaign(id: String?) {
        requireNotNull(id) { "id required" }
        service.softDelete(id)
    }
}
